### ISSUE TITLE
changes (cleanout2 branch)

### ISSUE THREAD

**Issue by SEAPUNK** [SEAPUNK/uWebSockets#2]
- **done** Clean up nodejs directory, move around files so it's installable directly via git
- **done** Use package.json `files` property to decide what goes in the published package
- **done** Change the installation process to this:
  - **done** Check if you can load an existing native addon, if so, use that
  - **done** Check if you can build a native addon (Windows version check), if not, fast-fail
  - **done** Try to build the binary with node-gyp as a fallback (or fail the installation)
- **WIP** Changes the make scripts to build for the environment they're in, rather than try to build for all environments. This moves the responsibility of building for different environments to CIs.
- **WIP** Configure CIs (Appveyor for Windows, Travis for Mac & Linux) to build and test commits, as well as publish GitHub release draft addon binaries on pushed git tag.
- **WIP** Write publish script that on `npm publish`, prepares the npm package by downloading all the required native addons from the GitHub releases before publishing them, then reminding the publisher to edit the specific GitHub release, since it's currently a draft (relies on package.json version being the same as the git tag, unless we can figure out a better way to do this)
  - Should the publish script not bother to download the binaries if they already exist in the `nodejs/` directory, allowing the publisher of not having to use the whole CI/GitHub release process, if they need/want to do it manually? This is potentially error prone.
---

Unresolved issues:

- [ ] CI test suite. More specifically,
  - [ ] `npm test`, which will include linting tests and regular tests.
  - [ ] Primus integration tests
  - [ ] engine.io integration tests
  - [ ] other integration tests (?)
- [ ] What is extension.cpp? git blame indicates that it's for building uws directly into nodejs, but I wonder how it's done -- there is no documentation on it.
- [ ] Should we use "direct" makefiles to build the binary first before building with node-gyp on install? The challenge here is the downloading and extracting of the header tarball before we can even build.

**Comment by SEAPUNK** [comment]
cc @alexhultman: How does this sound so far? I'm reworking my original PR to do just what's stated above instead, some of which is mostly the same.

Adding of the `semistandard` linter and making the JS code conform to it will probably be a separate PR.

**Comment by SEAPUNK** [comment]
Once I get the "configure CIs" todo list done, I'll demonstrate to you what it looks like, and what the publish process will look like as well, so you can be the judge of whether it makes everyone's lives easier or not.
